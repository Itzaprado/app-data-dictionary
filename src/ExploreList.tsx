import * as React from "react"
import { Menu, MenuItem, MenuGroup } from "looker-lens"
import { ILookmlModel, ILookmlModelNavExplore } from "@looker/sdk"

export default ({
  models,
  onExploreSelected
}: {
  models: ILookmlModel[]
  onExploreSelected: (
    model: ILookmlModel,
    explore: ILookmlModelNavExplore
  ) => void
}) => (
  <Menu>
    {models
      .filter(model => model.explores.length > 0)
      .map(model => {
        return (
          <MenuGroup label={model.label} key={model.name}>
            {model.explores.map(explore => {
              return (
                <MenuItem
                  key={explore.name}
                  onClick={() => onExploreSelected(model, explore)}
                >
                  {explore.label}
                </MenuItem>
              )
            })}
          </MenuGroup>
        )
      })}
  </Menu>
)
