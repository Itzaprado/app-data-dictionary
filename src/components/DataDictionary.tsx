/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React, { PureComponent, useState } from "react";
import {
  Chip,
  Flex,
  FlexItem,
  Heading,
  Spinner,
  theme,
} from "@looker/components";
import humanize from 'humanize-string'
import styled, { ThemeProvider } from "styled-components";
import { useAllModels } from "../utils/fetchers";
import "./styles.css";
import { PanelFields } from "./PanelFields";
import SidebarToggle from "./SidebarToggle";
import Spirals from "../images/spirals.png";
import "./styles.css";
import { useCurrentModel, useCurrentExplore } from "../utils/routes"
import { ColumnDescriptor } from "./interfaces";
import { SQLSnippet } from "./SQLSnippet";
import { Sidebar } from './Sidebar'
import { SidebarStyleProps } from "./interfaces";
import { NoModelsAvailable } from "./NoModelsAvailable";

export const columns: ColumnDescriptor[] = [
  {
    name: 'field-label',
    label: 'Field Label',
    rowValueDescriptor: 'label_short',
    formatter: (x: any) => x,
  },
  {
    name: 'description',
    label: 'Description',
    rowValueDescriptor: 'description',
    formatter: (x: any) => x,
  },
  {
    name: 'lookml-name',
    label: 'LookML Name',
    rowValueDescriptor: 'name',
    formatter: (x: any) => x,
  },
  {
    name: 'type',
    label: 'Type',
    rowValueDescriptor: 'type',
    formatter: (x: any) => humanize(x),
  },
  {
    name: 'sql',
    label: 'SQL',
    rowValueDescriptor: 'sql',
    formatter: (x: any) => <SQLSnippet src={x} />
  },
  {
    name: 'tags',
    label: 'Tags',
    rowValueDescriptor: 'tags',
    formatter: (tags: any) => {
      return tags.map((tag: string) => (
        <Chip disabled>{tag}</Chip>
      ))
    },
  },
]

export const DataDictionary: React.FC<{}> = () => {
  const unfilteredModels = useAllModels()
  const currentModel = useCurrentModel()
  const [sidebarOpen, setSidebarOpen] = React.useState(true)
  const [search, setSearch] = React.useState('')
  const { currentExplore, loadingExplore } = useCurrentExplore()

  let models

  if (unfilteredModels) {
    models = unfilteredModels.filter(m => m.explores && m.explores.some(e => !e.hidden))
    if (!models.length) {
      return <NoModelsAvailable />
    }
  }

  if (!models) {
    return <Flex alignItems="center" height="100%" justifyContent="center"><Spinner /></Flex>
  }

  const toggleFn = () => setSidebarOpen(!sidebarOpen);

  return (
    <ThemeProvider theme={theme}>
      <PageHeader style={{ backgroundImage: "url(" + Spirals + ")" }}>
        <FlexItem>
          <Heading as="h1" fontSize="xlarge" fontWeight="semiBold" mb="xsmall">
            Data Dictionary
          </Heading>
        </FlexItem>
      </PageHeader>
      <PageLayout open={sidebarOpen}>
        <LayoutSidebar>
          {sidebarOpen && <Sidebar
            currentExplore={currentExplore}
            currentModel={currentModel}
            loadingExplore={loadingExplore}
            models={models}
            search={search}
            setSearch={setSearch}
          />}
        </LayoutSidebar>
        <SidebarDivider open={sidebarOpen}>
          <SidebarToggle
            isOpen={sidebarOpen}
            onClick={toggleFn}
            headerHeight="114px"
          />
        </SidebarDivider>
        <PageContent>
          <PanelFields
            columns={columns}
            currentExplore={currentExplore}
            loadingExplore={loadingExplore}
            model={currentModel}
          />
        </PageContent>
      </PageLayout>
    </ThemeProvider>
  );
}


const PageHeader = styled(Flex)`
  background-color: ${theme.colors.palette.purple400};
  background-position: 100% 0;
  background-repeat: no-repeat;
  background-size: 836px 120px;
  padding: ${theme.space.large};

  h1 {
    color: ${theme.colors.palette.white};
  }
`;

const PageLayout = styled.div<SidebarStyleProps>`
  display: grid;
  grid-template-rows: 1fr;
  grid-template-columns: ${({ open }) =>
    open ? "16.625rem 0 1fr" : "1.5rem 0 1fr"};
  grid-template-areas: "sidebar divider main";
  position: relative;
`;

const PageContent = styled.div`
  grid-area: main;
  position: relative;
`;

const LayoutSidebar = styled.aside`
  position: absolute;
  top: 0;
  bottom: 0;
  width: 16.625rem;
  grid-area: sidebar;
  z-index: 0;
`;

const SidebarDivider = styled.div<SidebarStyleProps>`
  transition: border 0.3s;
  border-left: 1px solid
    ${({ theme, open }) =>
      open ? theme.colors.palette.charcoal200 : "transparent"};
  grid-area: divider;
  overflow: visible;
  position: relative;
  &:hover {
    border-left: 1px solid
      ${({ theme, open }) =>
        open
          ? theme.colors.palette.charcoal300
          : theme.colors.palette.charcoal200};
  }
`;

