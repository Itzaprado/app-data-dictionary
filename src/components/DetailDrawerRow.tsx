import React from "react";
import {
  Code,
  TableRow,
  TableDataCell,
} from "@looker/components";

import { ILookmlModelExploreField } from "@looker/sdk";
import { ColumnDescriptor } from "./interfaces";

export const DetailDrawerRow: React.FC<{
  column: ColumnDescriptor,
  field: ILookmlModelExploreField,
}> = ({ column, field }) => {
  return (
    <TableRow key={column.rowValueDescriptor}>
      <TableDataCell
        color="palette.charcoal700"
        p="medium"
        pl="small"
        pr="small"
      >
        { column.label }
      </TableDataCell>
      <TableDataCell
        className="break"
        color="palette.charcoal700"
        p="medium"
        pl="small"
        pr="small"
      >
        <Code
          color="palette.charcoal700"
          fontSize="small"
          className="break"
        >
          {/* Sorry, I promise the value is there.
  // @ts-ignore */}
          {column.formatter(field[column.rowValueDescriptor])}
        </Code>
      </TableDataCell>
    </TableRow>
  )
};
