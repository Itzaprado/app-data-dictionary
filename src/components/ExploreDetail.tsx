import React from "react"
import { ILookmlModel, ILookmlModelExplore } from "@looker/sdk/dist/sdk/models"
import { PageDetail } from "../components-generalized/Page"
import {
  MetadataItem,
  MetadataList
} from "../components-generalized/MetadataList"
import { Flex, FlexItem, Button } from "looker-lens/dist"
import { ExternalLink } from "../extract-to-framework/ExtensionLink"
import { exploreURL } from "../utils/urls"
import { FieldName } from "./ExploreFieldGrid"
import { Tags } from "../components-generalized/Tags"

interface ExploreDetailProps {
  model: ILookmlModel
  explore: ILookmlModelExplore
  onClose: () => void
}

export const ExploreDetail: React.FC<ExploreDetailProps> = ({
  explore,
  onClose
}) => {
  return (
    <PageDetail title={explore.label} onClose={onClose}>
      <MetadataList>
        <MetadataItem label="LookML Name">
          <FieldName>{explore.name}</FieldName>
        </MetadataItem>
        <MetadataItem label="Connection">
          <FieldName>{explore.connection_name}</FieldName>
        </MetadataItem>
        {explore.hidden && <MetadataItem label="Hidden">Yes</MetadataItem>}
        {explore.description && (
          <MetadataItem label="Description">{explore.description}</MetadataItem>
        )}
        {explore.tags && explore.tags.length > 0 ? (
          <MetadataItem label="Tags">
            <Tags tags={explore.tags} />
          </MetadataItem>
        ) : (
          undefined
        )}
        <MetadataItem compact label="Dimensions">
          {explore.fields.dimensions.length.toLocaleString()}
        </MetadataItem>
        <MetadataItem compact label="Measures">
          {explore.fields.measures.length.toLocaleString()}
        </MetadataItem>
        <MetadataItem compact label="Parameters">
          {explore.fields.parameters.length.toLocaleString()}
        </MetadataItem>
        <MetadataItem compact label="Filter-Only Fields">
          {explore.fields.filters.length.toLocaleString()}
        </MetadataItem>
        <MetadataItem compact label="Joins">
          {explore.joins.length.toLocaleString()}
        </MetadataItem>
      </MetadataList>
      <Flex>
        <FlexItem flex="1 1 auto" textAlign="center">
          <ExternalLink target="_blank" href={exploreURL(explore)}>
            <Button iconBefore="Explore" variant="transparent" size="xsmall">
              Explore
            </Button>
          </ExternalLink>
        </FlexItem>
      </Flex>
    </PageDetail>
  )
}
