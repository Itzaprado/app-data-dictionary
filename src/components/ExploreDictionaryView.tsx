import React from "react"
import {
  ILookmlModelExplore,
  ILookmlModelExploreField,
  ILookmlModel
} from "@looker/sdk/dist/sdk/models"
import { Text, Button, Box, Divider } from "looker-lens"
import { exploreURL } from "../utils/urls"
import ExploreFieldGrid from "./ExploreFieldGrid"
import {
  Page,
  PageHeader,
  PageHeaderTitle,
  PageHeaderControls,
  PageMasterDetail,
  PageMaster,
  PageDetail
} from "../components-generalized/Page"
import { ViewCustomizer } from "./ViewCustomizer"
import { FieldDetail, Tags } from "./FieldDetail"
import { ExternalLink } from "../extract-to-framework/ExtensionLink"

interface ExploreDictionaryViewState {
  detailField?: ILookmlModelExploreField
}

interface ExploreDictionaryViewProps {
  model: ILookmlModel
  explore: ILookmlModelExplore
}

export default class ExploreDictionaryView extends React.Component<
  ExploreDictionaryViewProps,
  ExploreDictionaryViewState
> {
  constructor(props: ExploreDictionaryViewProps) {
    super(props)
    this.state = {}
    this.setDetailField = this.setDetailField.bind(this)
  }

  setDetailField(detailField: ILookmlModelExploreField) {
    this.setState({ detailField })
  }

  render() {
    return (
      <Page>
        <PageHeader>
          <PageHeaderTitle> {this.props.explore.label}</PageHeaderTitle>
          <PageHeaderControls>
            <ViewCustomizer />
            <ExternalLink target="_blank" href={exploreURL(this.props.explore)}>
              <Button
                iconBefore="Explore"
                ml="large"
                variant="outline"
                size="small"
              >
                Explore
              </Button>
            </ExternalLink>
          </PageHeaderControls>
        </PageHeader>
        <PageMasterDetail>
          <PageMaster>
            <ExploreHeader explore={this.props.explore} />
            <ExploreFieldGrid
              {...this.props}
              setDetailField={this.setDetailField}
            />
          </PageMaster>
          {this.state.detailField && (
            <PageDetail
              title={this.state.detailField.label}
              onClose={() => this.setState({ detailField: undefined })}
            >
              <FieldDetail
                field={this.state.detailField}
                model={this.props.model}
                explore={this.props.explore}
              />
            </PageDetail>
          )}
        </PageMasterDetail>
      </Page>
    )
  }
}

const ExploreHeader = ({ explore }: { explore: ILookmlModelExplore }) => {
  const hasDescription = !!explore.description
  const hasTags = !!(explore.tags && explore.tags.length > 0)
  if (!hasDescription && !hasTags) {
    return <></>
  }
  return (
    <>
      {hasDescription && (
        <Text mb="medium" variant="subdued">
          {explore.description}
        </Text>
      )}
      {hasTags && (
        <Box mb="medium">
          <Tags tags={explore.tags!} />
        </Box>
      )}
      <Divider my="medium" appearance="light" />
    </>
  )
}
