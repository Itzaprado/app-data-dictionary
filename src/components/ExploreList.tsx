import React, { PureComponent, useState } from "react";
import styled from "styled-components";
import {
  FlexItem,
  List,
  ListItem,
  MenuItem,
  theme,
} from "@looker/components";
import "./styles.css";
import { useCurrentExplore} from "../utils/routes";
import { internalExploreURL, useCurrentModel } from "../utils/routes"
import { useHistory } from "react-router"

export const ExploreList: React.FC<{search: String}> = ({ search }) => {
  const currentExplore = useCurrentExplore()
  const currentModel = useCurrentModel()
  const history = useHistory()
  return (
    <FlexItem pt="xlarge">
      <List>
        { currentModel && currentModel.explores.map((explore: any) => {
          if (!explore.hidden && (!search || explore.label.toLowerCase().includes(search.toLowerCase()))) {
            return (
              <ListItem mb="0" key={explore.name}>
                <CustomLink
                  onClick={() => {
                    history.push(
                      internalExploreURL({
                        model: currentModel.name,
                        explore: explore.name
                      })
                    )
                  }}
                  className={currentExplore && explore.name === currentExplore.name ? "active" : null}
                >
                  {explore.label}
                </CustomLink>
              </ListItem>
            )
          }
        }) }
      </List>
    </FlexItem>
  )
}

const CustomLink = styled(MenuItem)`
  color: ${theme.colors.palette.purple700};
  display: block;
  transition: all 0.3s ease;

  &.active,
  &:hover,
  &:focus {
    background-color: ${theme.colors.palette.purple000};
    color: ${theme.colors.palette.purple400};
    text-decoration: none;
  }
`;
