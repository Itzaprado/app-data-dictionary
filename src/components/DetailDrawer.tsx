import React from "react";
import {
  theme,
  ButtonTransparent,
  DrawerManager,
  Flex,
  FlexItem,
  Heading,
  ModalContent,
  Paragraph,
  Table,
  TableRow,
  TableDataCell,
  TableBody
} from "@looker/components";
import styled from "styled-components";

import {ILookmlModel, ILookmlModelExplore, ILookmlModelExploreField} from "@looker/sdk";
import {ColumnDescriptor} from "./interfaces";
import { ExternalLink } from "./ExternalLink";
import { exploreFieldURL } from "../utils/urls";
import { canGetDistribution, canGetTopValues } from "../utils/queries";
import { QueryChart } from './QueryChart'
import { DetailDrawerRow } from "./DetailDrawerRow";


// Page Header
const TableRowCustom = styled(TableRow)`
  transition: background-color 0.3s ease;
  &.active,
  &:hover {
    background-color: ${theme.colors.palette.charcoal100};
  }

  td {
    word-break: break-word;
  }
`;

export const DetailDrawer: React.FC<{
  columns: ColumnDescriptor[],
  explore: ILookmlModelExplore,
  model: ILookmlModel,
  field: ILookmlModelExploreField,
  shownColumns: string[],
}> = ({ columns, explore, field, model, shownColumns }) => {
  return (
    <DrawerManager
      content={
        <ModalContent>
          <Heading as="h3" fontWeight="semiBold">
            {field.label_short}
          </Heading>
          <Flex flexDirection="column">
            <FlexItem pt="large" pb="xlarge">
              <Paragraph>{field.description}</Paragraph>
            </FlexItem>
            <FlexItem>
              <Heading
                as="h4"
                fontSize="small"
                fontWeight="semiBold"
                mb="small"
              >
                About this Field
              </Heading>
            </FlexItem>
            <FlexItem pb="xlarge">
              <Table width="100%">
                <TableBody fontSize="small">
                  { columns.map(column => {
                    return (
                      <DetailDrawerRow
                        key={column.rowValueDescriptor}
                        column={column}
                        field={field}
                      />
                    )
                  })}
                </TableBody>
              </Table>
            </FlexItem>

            {canGetDistribution({
              model,
              explore,
              field
            }) && (
              <QueryChart
                type={{
                  type: "Distribution",
                  model,
                  explore,
                  field
                }}
              />
            )}

            {canGetTopValues({
              model,
              explore,
              field
            }) && (
              <QueryChart
                type={{
                  type: "Values",
                  model,
                  explore,
                  field
                }}
              />
            )}

            <FlexItem
              borderTop={`1px solid ${
                theme.colors.palette.charcoal200
                }`}
              pb="xlarge"
              pt="xlarge"
            >
              <Flex alignItems="center" justifyContent="center">
                <ExternalLink target="_blank" href={field.lookml_link}>
                  <ButtonTransparent
                    mr="small"
                    ml="small"
                    iconBefore="LogoRings"
                  >
                    Go to LookML
                  </ButtonTransparent>
                </ExternalLink>


                <ExternalLink target="_blank" href={exploreFieldURL(explore, field)}>
                  <ButtonTransparent
                    mr="small"
                    ml="small"
                    iconBefore="Explore"
                  >
                    Explore with Field
                  </ButtonTransparent>
                </ExternalLink>


              </Flex>
            </FlexItem>
          </Flex>
        </ModalContent>
      }
    >
      {onClick => (
        <TableRowCustom onClick={onClick}>
          { columns.map(column => {
            if (shownColumns.includes(column.rowValueDescriptor)) {
              return (
                <TableDataCell
                  color="palette.charcoal700"
                  p="medium"
                  pl="small"
                  key={column.rowValueDescriptor}
                >
                  {/* This is bad, sorry, I promise the value is there.
                    // @ts-ignore */}
                  {column.formatter(field[column.rowValueDescriptor])}
                </TableDataCell>
              )
            }
          })}
        </TableRowCustom>
      )}
    </DrawerManager>
  );
};
