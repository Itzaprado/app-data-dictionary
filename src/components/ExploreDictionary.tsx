import React from "react"
import { ExploreDictionaryView } from "./ExploreDictionaryView"
import PlainPageLoading from "../components-generalized/PlainPageLoading"
import { useCurrentExplore, useCurrentModel } from "../utils/routes"

export const ExploreDictionary: React.FC = props => {
  const currentModel = useCurrentModel()
  const currentExplore = useCurrentExplore()
  if (currentExplore && currentModel) {
    return (
      <ExploreDictionaryView explore={currentExplore} model={currentModel} />
    )
  } else {
    return <PlainPageLoading />
  }
}
