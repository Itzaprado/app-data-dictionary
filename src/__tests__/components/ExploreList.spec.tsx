import React from 'react';
import renderer from 'react-test-renderer';
import { mockCurrentModel, mockCurrentExplore} from "../MockData/MockData";
import { theme} from "@looker/components"
import { ThemeProvider } from "styled-components"

import { ExploreList } from '../../components/ExploreList'

jest.mock('../../utils/routes', () => {
  return {
    useCurrentModel: jest.fn(() => {
      return mockCurrentModel
    }),
    useCurrentExplore: jest.fn(() => {
      return mockCurrentExplore
    })
  }
})

jest.mock('react-router', () => {
  return {
    useHistory: jest.fn(() => {
      push: () => {}
    })
  }
})

jest.mock("../../components/PanelFields", () => ({
  PanelFields: () => "PanelFields"
}))

jest.mock("../../components/Sidebar", () => ({
  Sidebar: () => "Sidebar"
}))

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ExploreList
          search={null}
        />)
      </ThemeProvider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
})
