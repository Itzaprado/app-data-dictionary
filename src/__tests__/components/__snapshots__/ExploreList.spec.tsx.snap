// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="FlexItem-sc-1re6mwg fxCVEF"
  >
    <ul
      className="List-sc-1dd290l VHTwG"
      type="none"
    >
      <li
        className="ListItem-sc-16mi6gz ejtiuJ"
      >
        <li
          bg="#FFFFFF"
          className="MenuItemListItem-sc-t4lwjd dOUswS sc-bdVaJa gjuDgd"
          color="#4C535B"
          fontSize="small"
          fontWeight="normal"
          onBlur={[Function]}
          onClick={[Function]}
          onKeyUp={[Function]}
        >
          <button
            className="MenuItemButton-sc-f5mc0o iUjhjR"
            role="menuitem"
          >
            AdWord Impressions
          </button>
        </li>
      </li>
      <li
        className="ListItem-sc-16mi6gz ejtiuJ"
      >
        <li
          bg="#FFFFFF"
          className="MenuItemListItem-sc-t4lwjd dOUswS sc-bdVaJa gjuDgd"
          color="#4C535B"
          fontSize="small"
          fontWeight="normal"
          onBlur={[Function]}
          onClick={[Function]}
          onKeyUp={[Function]}
        >
          <button
            className="MenuItemButton-sc-f5mc0o iUjhjR"
            role="menuitem"
          >
            AdWord Impressions by Ad Group
          </button>
        </li>
      </li>
      <li
        className="ListItem-sc-16mi6gz ejtiuJ"
      >
        <li
          bg="#FFFFFF"
          className="MenuItemListItem-sc-t4lwjd dOUswS sc-bdVaJa gjuDgd"
          color="#4C535B"
          fontSize="small"
          fontWeight="normal"
          onBlur={[Function]}
          onClick={[Function]}
          onKeyUp={[Function]}
        >
          <button
            className="MenuItemButton-sc-f5mc0o iUjhjR"
            role="menuitem"
          >
            AdWord Impressions by Keyword
          </button>
        </li>
      </li>
      <li
        className="ListItem-sc-16mi6gz ejtiuJ"
      >
        <li
          bg="#FFFFFF"
          className="MenuItemListItem-sc-t4lwjd dOUswS sc-bdVaJa gjuDgd"
          color="#4C535B"
          fontSize="small"
          fontWeight="normal"
          onBlur={[Function]}
          onClick={[Function]}
          onKeyUp={[Function]}
        >
          <button
            className="MenuItemButton-sc-f5mc0o iUjhjR"
            role="menuitem"
          >
            AdWord Impressions by Geo
          </button>
        </li>
      </li>
      <li
        className="ListItem-sc-16mi6gz ejtiuJ"
      >
        <li
          bg="#FFFFFF"
          className="MenuItemListItem-sc-t4lwjd dOUswS sc-bdVaJa gjuDgd"
          color="#4C535B"
          fontSize="small"
          fontWeight="normal"
          onBlur={[Function]}
          onClick={[Function]}
          onKeyUp={[Function]}
        >
          <button
            className="MenuItemButton-sc-f5mc0o iUjhjR"
            role="menuitem"
          >
            AdWord Impressions by Age Range
          </button>
        </li>
      </li>
      <li
        className="ListItem-sc-16mi6gz ejtiuJ"
      >
        <li
          bg="#FFFFFF"
          className="MenuItemListItem-sc-t4lwjd dOUswS sc-bdVaJa gjuDgd"
          color="#4C535B"
          fontSize="small"
          fontWeight="normal"
          onBlur={[Function]}
          onClick={[Function]}
          onKeyUp={[Function]}
        >
          <button
            className="MenuItemButton-sc-f5mc0o iUjhjR"
            role="menuitem"
          >
            AdWord Impressions by Audience
          </button>
        </li>
      </li>
      <li
        className="ListItem-sc-16mi6gz ejtiuJ"
      >
        <li
          bg="#FFFFFF"
          className="MenuItemListItem-sc-t4lwjd dOUswS sc-bdVaJa gjuDgd"
          color="#4C535B"
          fontSize="small"
          fontWeight="normal"
          onBlur={[Function]}
          onClick={[Function]}
          onKeyUp={[Function]}
        >
          <button
            className="MenuItemButton-sc-f5mc0o iUjhjR"
            role="menuitem"
          >
            AdWord Impressions by Video
          </button>
        </li>
      </li>
    </ul>
  </div>,
  ")",
]
`;
